androidApp/main/app/src/main/java/com/simplecityapps/shuttle/ui/screens/library/albums/detail/AlbumDetailPresenter.kt

```diff
@@ -111,8 +109,8 @@ class AlbumDetailPresenter @AssistedInject constructor(
     }
 
     override fun onQueuePositionChanged(oldPosition: Int?, newPosition: Int?) {
-        getCurrentSong()?.let {
-            view?.onCurrentSongChanged(it)
+        getCurrentSong()?.let { currentSong ->
+            view?.setData(this@AlbumDetailPresenter.songs, currentSong)
```

Comment:
Out of curiosity. Why is the qualifier needed in `this@AlbumDetailPresenter.songs`? I've tried and it seems to work without it.

By the way, nice work! These changes make it much simpler. Thanks!

Reply:
Oh, `this@` is not needed, that looks like a copy paste error :D